src/commands/validate.ts

```diff
@@ -29,18 +30,30 @@ export default class Validate extends Command {
       specWatcher({spec: specFile, handler: this, handlerName: 'validate'});
     }
     try {
+      const specFile = await load(filePath);
+      if (watchMode) {
+        specWatcher({ spec: specFile, handler: this, handlerName: 'validate' });
+      }
+
       if (specFile.getFilePath()) {
         await parser.parse(specFile.text());
         this.log(`File ${specFile.getFilePath()} successfully validated!`);
       } else if (specFile.getFileURL()) {
         await parser.parse(specFile.text());
         this.log(`URL ${specFile.getFileURL()} successfully validated`);
       }
-    } catch (error) {
-      throw new ValidationError({
-        type: 'parser-error',
-        err: error
-      });
+    } catch (e) {
+      if (e instanceof SpecificationFileNotFound) {
+        this.error(new ValidationError({
+          type: 'invalid-file',
+          filepath: filePath
+        }));
```

Comment:
This is what I meant in my previous comment:

```suggestion
        this.error(new SpecificationFileNotFound(filePath));
```

But since `e` is already a `SpecificationFileNotFound` error, you can simply forward the error as is:

```suggestion
        this.error(e);
```